packages/@react-aria/button/src/useButton.ts

```diff
@@ -87,6 +87,15 @@ export function useButton(props: AriaButtonProps<ElementType>, ref: RefObject<an
     ref
   });
 
+  // don't apply type button to non button elements
+  if (elementType !== 'button' && type === 'button') {
+    type = undefined;
+  }
+  // don't apply disabled to non button/input elements
+  if (elementType !== 'input' && elementType !== 'button') {
+    isDisabled = undefined;
+  }
```

Comment:
Seems like maybe we need to change the above to something like

```js
if (elementType === 'button') {
  additionalProps = {
    type,
    disabled: isDisabled
  };
}
```

and remove those props from below?